src/app/client/src/app/app.component.ts

```diff
@@ -174,9 +175,15 @@ export class AppComponent implements OnInit, OnDestroy {
     this.appId = this.userService.appId;
   }
 
+  handleLocatoionPopupInSignup() {
+    this.hideLocatoionPopupInSignup = _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.data.hideLocatoionPopup') ||
+          _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.firstChild.data.hideLocatoionPopup');
+  }
+
   isLocationStatusRequired() {
     const url = this.router.url;
-    return !!(_.includes(url, 'signup') || _.includes(url, 'recover') || _.includes(url, 'sign-in'));
+    this.handleLocatoionPopupInSignup();
+    return !!(_.includes(url, 'signup') || _.includes(url, 'recover') || _.includes(url, 'sign-in') || this.hideLocatoionPopupInSignup);
```

Comment:
not sure if these both ways of checking is required ?

Reply:
Since `this.router.url;` is coming undefined if I directly hit on `/signup` so it requires both checks